File: py-appscript/trunk/py-appscript-0.19.0/test/test_mactypes.py
#!/usr/local/bin/python

import unittest, os, os.path
from aem.ae import MacOSError
import mactypes

class TC_MacTypes(unittest.TestCase):

	dir = '/private/tmp'
	
	def setUp(self):
		self.path1 = '/private' + os.tempnam(self.dir, 'py-mactypes-test.') # tempnam raises a security warning re. security; it's part of the test code, not mactypes, so ignore it
		file(self.path1, 'w').close()
		fname = os.path.split(self.path1)[1]
		self.path2 = os.path.join(self.dir, 'moved-' + fname)
		# print "path: %r" % self.path1 # e.g. /private/tmp/py-mactypes-test.VLrUW7
	
	def test_alias(self):
		# make alias
		self.f = mactypes.Alias(self.path1)
		
		self.assertEqual("mactypes.Alias(u%r)" % self.path1,  repr(self.f))
		
		#print "alias path 1: %s" % f.path # e.g. /private/tmp/py-mactypes-test.VLrUW7
		self.assertEqual(self.path1, self.f.path)
		
		# get desc
		#print `f.desc.type, f.desc.data` # alis, [binary data]
		self.assertEqual('alis', self.f.desc.type)

		
		# check alias keeps track of moved file
		os.rename(self.path1, self.path2)
		# print "alias path 2: %r" % f.path # /private/tmp/moved-py-mactypes-test.VLrUW7
		self.assertEqual(self.path2, self.f.path)

		self.assertEqual("mactypes.Alias(u%r)" % self.path2, repr(self.f))
		
		# check a FileNotFoundError is raised if getting path/FileURL for a filesystem object that no longer exists
		os.remove(self.path2)
		self.assertRaises(MacOSError, lambda:self.f.path) # File not found.
		self.assertRaises(MacOSError, lambda:self.f.file) # File not found.


	def test_fileURL(self):

		g = mactypes.File('/non/existent path')

		self.assertEqual('/non/existent path', g.path)
		
		self.assertEqual('furl', g.desc.type)
		self.assertEqual('file://localhost/non/existent%20path', g.desc.data)

		self.assertEqual("mactypes.File(u'/non/existent path')", repr(g.file))

		# check a not-found error is raised if getting Alias for a filesystem object that doesn't exist
		self.assertRaises(MacOSError, lambda:g.alias) # File "/non/existent path" not found.


if __name__ == '__main__':
	unittest.main()
	

File: rb-appscript/trunk/test/test_mactypes.rb
#!/usr/bin/ruby -w

begin; require 'rubygems'; rescue LoadError; end

require 'test/unit'
require "_aem/mactypes"


class TC_MacTypes < Test::Unit::TestCase
	
	def setup
		@path1 = `mktemp -t codecs-test`.chomp
		dir, fname = File.split(@path1)
		@path2 = File.join(dir, 'moved-' + fname)
		# puts "path: #{@path1}" # e.g. /tmp/codecs-test.HWr1EnE3
	end
	
	def normalize(path)
		return path.gsub(/\/+/, '/') # quick-n-dirty
	end
	
	def test_alias
		# make alias
		f = MacTypes::Alias.path(@path1)
		
		
		assert_equal(MacTypes::Alias.path(@path1), f)
		
		# get path
		# note that initial /tmp/codecs-test... path will automatically change to /private/tmp/codecs-test...
		p1 = normalize('/private'+@path1)
		p2 = normalize('/private'+@path2)
				
		assert_equal(p1, normalize(f.path))
		
		#puts "alias path 1: #{f}" # e.g. /private/tmp/codecs-test.HWr1EnE3
		assert_equal(p1, f.to_s)
		
		# get desc
		#puts f.desc.type, f.desc.data # alis, [binary data]
		assert_equal('alis', f.desc.type)

		
		# check alias keeps track of moved file
		`mv #{@path1} #{@path2}`
		# puts "alias path 2: #{f}" # /private/tmp/moved-codecs-test.HWr1EnE3
		assert_equal(p2, f.to_s)

		assert_equal(p2, normalize(f.path))
		
		# check a FileNotFoundError is raised if getting path/FileURL for a filesystem object that no longer exists
		`rm #{@path2}`
		assert_raises(MacTypes::FileNotFoundError) { f.to_s } # File not found.
		assert_raises(MacTypes::FileNotFoundError) { f.to_file_url } # File not found.
		
		assert_equal(MacTypes::Alias.path("/Library/Scripts/"), MacTypes::Alias.path("/Library/Scripts/"))
		assert_not_equal(MacTypes::Alias.path("/Library/Scripts/"), MacTypes::Alias.path("/Applications/"))
	end


	def test_file_url

		g = MacTypes::FileURL.path('/non/existent path')

		assert_equal('/non/existent path', g.to_s)
		
		assert_equal('furl', g.desc.type)
		assert_equal('file://localhost/non/existent%20path', g.desc.data)

		assert_equal('MacTypes::FileURL.path("/non/existent path")', g.to_file_url.inspect)

		# check a not-found error is raised if getting Alias for a filesystem object that doesn't exist
		assert_raises(MacTypes::FileNotFoundError) { g.to_alias } # File "/non/existent path" not found.
		
		assert_equal(MacTypes::FileURL.path("/Library/Scripts/"), MacTypes::FileURL.path("/Library/Scripts/"))
		assert_not_equal(MacTypes::FileURL.path("/Library/Scripts/"), MacTypes::FileURL.path("/Applications/"))
		assert_not_equal(MacTypes::FileURL.path("/Library/Scripts/"), MacTypes::Alias.path("/Library/Scripts/"))
	end
end